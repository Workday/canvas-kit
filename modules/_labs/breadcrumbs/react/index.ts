import Breadcrumbs from './lib/Breadcrumbs';

export default Breadcrumbs;
export {Breadcrumbs};
export * from './lib/Breadcrumbs';
