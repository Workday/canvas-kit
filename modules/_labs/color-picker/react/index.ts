export {default, default as ColorPicker, ColorPickerProps} from './lib/ColorPicker';
export {Swatch} from './lib/parts/Swatch';
