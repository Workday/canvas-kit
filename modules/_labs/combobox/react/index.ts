import Combobox from './lib/Combobox';

export default Combobox;
export {Combobox};
export * from './lib/Combobox';
