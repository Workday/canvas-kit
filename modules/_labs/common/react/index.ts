export * from './lib/Flex';
export * from './lib/Box';
