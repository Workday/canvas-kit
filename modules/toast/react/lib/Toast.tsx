import * as React from 'react';
import Popup, {PopupPadding} from '@workday/canvas-kit-react-popup';
import {spacing, colors, type, CanvasColor} from '@workday/canvas-kit-react-core';
import {SystemIcon} from '@workday/canvas-kit-react-icon';
import {checkIcon} from '@workday/canvas-system-icons-web';
import {CanvasSystemIcon} from '@workday/design-assets-types';
import {TransformOrigin} from '@workday/canvas-kit-react-common';
import styled from '@emotion/styled';

export interface ToastProps {
  /**
   * The icon of the Toast.
   * @default checkIcon
   */
  icon?: CanvasSystemIcon;
  /**
   * The color of the Toast icon.
   * @default colors.greenApple400
   */
  iconColor?: CanvasColor | string; // TODO: Fix
  /**
   * The text of the Toast message.
   */
  children: string;
  /**
   * The function called when the Toast is closed.
   */
  onClose?: () => void;
  /**
   * The function called when the Toast action is clicked.
   */
  onActionClick?: () => void;
  /**
   * The text of the Toast action.
   */
  actionText?: string;
  /**
   * The origin from which the Toast will animate.
   * @default {horizontal: 'center', vertical: 'top'}
   */
  transformOrigin?: TransformOrigin;
}
const toastWidth = 360;

const ToastContentContainer = styled('div')<Pick<ToastProps, 'onClose'>>(
  {
    display: 'flex',
    alignItems: 'center',
    ...type.body2,
  },
  ({onClose}) => ({
    marginRight: onClose ? spacing.m : undefined,
  })
);

const ToastSystemIcon = styled(SystemIcon)({
  marginRight: spacing.s,
  alignSelf: 'start',
});

const ActionButton = styled('button')({
  display: 'block',
  backgroundColor: 'transparent', // To prevent Safari from rendering grey 'buttonface' as bgcolor
  border: 'none',
  padding: 0,
  marginTop: spacing.xxxs,
  ...type.body2,
  ...type.variant.link,
});

const Message = styled('div')({
  wordBreak: 'break-word',
  wordWrap: 'break-word',
});

export default class Toast extends React.Component<ToastProps> {
  public render() {
    const {
      icon = checkIcon as CanvasSystemIcon, // needed for TS2742 - https://github.com/microsoft/TypeScript/issues/29808
      iconColor = colors.greenApple400,
      transformOrigin,
      onClose,
      onActionClick,
      actionText,
      ...elemProps
    } = this.props;
    return (
      <Popup
        width={toastWidth}
        transformOrigin={transformOrigin}
        padding={PopupPadding.s}
        handleClose={onClose}
        closeIconSize="small"
        {...elemProps}
      >
        <ToastContentContainer onClose={onClose}>
          {icon && <ToastSystemIcon color={iconColor} colorHover={iconColor} icon={icon} />}
          <Message>
            {this.props.children}
            {onActionClick && <ActionButton onClick={onActionClick}>{actionText}</ActionButton>}
          </Message>
        </ToastContentContainer>
      </Popup>
    );
  }
}
