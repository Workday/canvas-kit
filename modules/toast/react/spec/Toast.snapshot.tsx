import * as React from 'react';
import Toast from '../lib/Toast';
import * as renderer from 'react-test-renderer';
import {checkIcon} from '@workday/canvas-system-icons-web';

describe('Toast Snapshots', () => {
  test('renders as expected', () => {
    const component = renderer.create(<Toast>Your data has been updated.</Toast>);
    expect(component).toMatchSnapshot();
  });
  test('renders with a close button', () => {
    const component = renderer.create(
      <Toast onClose={jest.fn()}>Your data has been updated.</Toast>
    );
    expect(component).toMatchSnapshot();
  });
  test('renders with an icon', () => {
    const component = renderer.create(
      <Toast icon={checkIcon} onClose={jest.fn()}>
        Success
      </Toast>
    );
    expect(component).toMatchSnapshot();
  });
  test('renders with an action link', () => {
    const component = renderer.create(
      <Toast actionText={'View More'} onActionClick={jest.fn()} onClose={jest.fn()}>
        Process failed.
      </Toast>
    );
    expect(component).toMatchSnapshot();
  });
});
