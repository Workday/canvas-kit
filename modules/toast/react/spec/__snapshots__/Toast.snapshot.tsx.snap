// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast Snapshots renders as expected 1`] = `
.emotion-8 {
  position: relative;
  width: 360px;
  -webkit-animation: animation-221pkp;
  animation: animation-221pkp;
  -webkit-animation-duration: 150ms;
  animation-duration: 150ms;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

.emotion-7 {
  background-color: #ffffff;
  border: 1px solid #ced3d9;
  border-radius: 3px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 16px;
  width: 360px;
}

.emotion-6 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 13px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-1 {
  display: inline-block;
  margin-right: 16px;
  -webkit-align-self: start;
  -ms-flex-item-align: start;
  align-self: start;
}

.emotion-1 svg {
  display: block;
}

.emotion-1 .wd-icon-fill {
  fill: #43c463;
}

.emotion-1:hover .wd-icon-fill {
  fill: #43c463;
}

.emotion-1 .wd-icon-accent {
  fill: #43c463;
}

.emotion-1:hover .wd-icon-accent {
  fill: #43c463;
}

.emotion-1 .wd-icon-background {
  fill: transparent;
}

.emotion-1:hover .wd-icon-background {
  fill: transparent;
}

.emotion-2 {
  word-break: break-word;
  word-wrap: break-word;
}

<div
  className="emotion-8"
  role="dialog"
  width={360}
>
  <div
    className="emotion-7"
    width={360}
  >
    <div
      className="emotion-6"
    >
      <div
        className="emotion-4 emotion-5"
      >
        <span
          className="emotion-0 emotion-1"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-check wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M8.657 15.85l10.26-10.26a.49.49 0 0 1 .703.01l.7.7a.495.495 0 0 1 .01.704L9.005 18.33a.488.488 0 0 1-.692.002L8.3 18.32l-4.946-4.946a.495.495 0 0 1 .003-.711l.7-.7a.507.507 0 0 1 .711-.004l3.889 3.89z\\" class=\\"wd-icon-fill\\"/></g></svg>",
            }
          }
        />
        <div
          className="emotion-2 emotion-3"
        >
          Your data has been updated.
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Toast Snapshots renders with a close button 1`] = `
.emotion-12 {
  position: relative;
  width: 360px;
  -webkit-animation: animation-221pkp;
  animation: animation-221pkp;
  -webkit-animation-duration: 150ms;
  animation-duration: 150ms;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

.emotion-11 {
  background-color: #ffffff;
  border: 1px solid #ced3d9;
  border-radius: 3px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 16px;
  width: 360px;
}

.emotion-10 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-5 {
  display: inline-block;
  margin-right: 16px;
  -webkit-align-self: start;
  -ms-flex-item-align: start;
  align-self: start;
}

.emotion-5 svg {
  display: block;
}

.emotion-5 .wd-icon-fill {
  fill: #43c463;
}

.emotion-5:hover .wd-icon-fill {
  fill: #43c463;
}

.emotion-5 .wd-icon-accent {
  fill: #43c463;
}

.emotion-5:hover .wd-icon-accent {
  fill: #43c463;
}

.emotion-5 .wd-icon-background {
  fill: transparent;
}

.emotion-5:hover .wd-icon-background {
  fill: transparent;
}

.emotion-6 {
  word-break: break-word;
  word-wrap: break-word;
}

.emotion-3 {
  position: absolute;
  right: 10px;
  top: 10px;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 13px;
  border-radius: 50%;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  border-width: 0px;
  background-color: transparent;
  margin: -6px;
  width: 32px;
  height: 32px;
}

.emotion-1:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-1:disabled,
.emotion-1:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-1 .wd-icon {
  display: inline-block;
  vertical-align: middle;
}

.emotion-1 span .wd-icon-fill,
.emotion-1 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #7b858f;
}

.emotion-1:focus {
  background-color: transparent;
}

.emotion-1:focus span .wd-icon-fill,
.emotion-1:focus span .wd-icon-accent {
  fill: #7b858f;
}

.emotion-1:hover:focus {
  background-color: transparent;
}

.emotion-1:active,
.emotion-1:focus:active,
.emotion-1:hover:active {
  background-color: transparent;
  border-color: transparent;
}

.emotion-1:active span .wd-icon-fill,
.emotion-1:focus:active span .wd-icon-fill,
.emotion-1:hover:active span .wd-icon-fill,
.emotion-1:active span .wd-icon-accent,
.emotion-1:focus:active span .wd-icon-accent,
.emotion-1:hover:active span .wd-icon-accent {
  fill: #333d47;
}

.emotion-1:hover {
  background-color: transparent;
}

.emotion-1:hover span .wd-icon-fill,
.emotion-1:hover span .wd-icon-accent {
  fill: #333d47;
}

.emotion-1:disabled,
.emotion-1:active:disabled,
.emotion-1:focus:disabled,
.emotion-1:hover:disabled {
  background-color: transparent;
}

.emotion-1:disabled span .wd-icon-fill,
.emotion-1:active:disabled span .wd-icon-fill,
.emotion-1:focus:disabled span .wd-icon-fill,
.emotion-1:hover:disabled span .wd-icon-fill,
.emotion-1:disabled span .wd-icon-accent,
.emotion-1:active:disabled span .wd-icon-accent,
.emotion-1:focus:disabled span .wd-icon-accent,
.emotion-1:hover:disabled span .wd-icon-accent {
  fill: #b9c0c7;
}

.emotion-1:not([disabled]):focus,
.emotion-1:not([disabled]):active {
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

.emotion-1 span svg {
  width: 20px;
  height: 20px;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-0 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 13px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  margin-right: 24px;
}

<div
  className="emotion-12"
  role="dialog"
  width={360}
>
  <div
    className="emotion-3"
  >
    <button
      aria-label="Close"
      className="emotion-1 emotion-2"
      onClick={[MockFunction]}
      title="Close"
    >
      <span
        className="emotion-0"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-x wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M5.996 4.582a.5.5 0 0 0-.717-.003l-.7.7a.504.504 0 0 0 .003.717L10.586 12l-6.004 6.004a.5.5 0 0 0-.003.717l.7.7c.197.197.514.2.717-.003L12 13.414l6.004 6.004a.5.5 0 0 0 .717.003l.7-.7a.504.504 0 0 0-.003-.717L13.414 12l6.004-6.004a.5.5 0 0 0 .003-.717l-.7-.7a.504.504 0 0 0-.717.003L12 10.586 5.996 4.582z\\" class=\\"wd-icon-fill\\"/></g></svg>",
          }
        }
      />
    </button>
  </div>
  <div
    className="emotion-11"
    width={360}
  >
    <div
      className="emotion-10"
    >
      <div
        className="emotion-8 emotion-9"
        onClose={[MockFunction]}
      >
        <span
          className="emotion-4 emotion-5"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-check wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M8.657 15.85l10.26-10.26a.49.49 0 0 1 .703.01l.7.7a.495.495 0 0 1 .01.704L9.005 18.33a.488.488 0 0 1-.692.002L8.3 18.32l-4.946-4.946a.495.495 0 0 1 .003-.711l.7-.7a.507.507 0 0 1 .711-.004l3.889 3.89z\\" class=\\"wd-icon-fill\\"/></g></svg>",
            }
          }
        />
        <div
          className="emotion-6 emotion-7"
        >
          Your data has been updated.
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Toast Snapshots renders with an action link 1`] = `
.emotion-14 {
  position: relative;
  width: 360px;
  -webkit-animation: animation-221pkp;
  animation: animation-221pkp;
  -webkit-animation-duration: 150ms;
  animation-duration: 150ms;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

.emotion-13 {
  background-color: #ffffff;
  border: 1px solid #ced3d9;
  border-radius: 3px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 16px;
  width: 360px;
}

.emotion-12 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-5 {
  display: inline-block;
  margin-right: 16px;
  -webkit-align-self: start;
  -ms-flex-item-align: start;
  align-self: start;
}

.emotion-5 svg {
  display: block;
}

.emotion-5 .wd-icon-fill {
  fill: #43c463;
}

.emotion-5:hover .wd-icon-fill {
  fill: #43c463;
}

.emotion-5 .wd-icon-accent {
  fill: #43c463;
}

.emotion-5:hover .wd-icon-accent {
  fill: #43c463;
}

.emotion-5 .wd-icon-background {
  fill: transparent;
}

.emotion-5:hover .wd-icon-background {
  fill: transparent;
}

.emotion-8 {
  word-break: break-word;
  word-wrap: break-word;
}

.emotion-3 {
  position: absolute;
  right: 10px;
  top: 10px;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 13px;
  border-radius: 50%;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  border-width: 0px;
  background-color: transparent;
  margin: -6px;
  width: 32px;
  height: 32px;
}

.emotion-1:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-1:disabled,
.emotion-1:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-1 .wd-icon {
  display: inline-block;
  vertical-align: middle;
}

.emotion-1 span .wd-icon-fill,
.emotion-1 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #7b858f;
}

.emotion-1:focus {
  background-color: transparent;
}

.emotion-1:focus span .wd-icon-fill,
.emotion-1:focus span .wd-icon-accent {
  fill: #7b858f;
}

.emotion-1:hover:focus {
  background-color: transparent;
}

.emotion-1:active,
.emotion-1:focus:active,
.emotion-1:hover:active {
  background-color: transparent;
  border-color: transparent;
}

.emotion-1:active span .wd-icon-fill,
.emotion-1:focus:active span .wd-icon-fill,
.emotion-1:hover:active span .wd-icon-fill,
.emotion-1:active span .wd-icon-accent,
.emotion-1:focus:active span .wd-icon-accent,
.emotion-1:hover:active span .wd-icon-accent {
  fill: #333d47;
}

.emotion-1:hover {
  background-color: transparent;
}

.emotion-1:hover span .wd-icon-fill,
.emotion-1:hover span .wd-icon-accent {
  fill: #333d47;
}

.emotion-1:disabled,
.emotion-1:active:disabled,
.emotion-1:focus:disabled,
.emotion-1:hover:disabled {
  background-color: transparent;
}

.emotion-1:disabled span .wd-icon-fill,
.emotion-1:active:disabled span .wd-icon-fill,
.emotion-1:focus:disabled span .wd-icon-fill,
.emotion-1:hover:disabled span .wd-icon-fill,
.emotion-1:disabled span .wd-icon-accent,
.emotion-1:active:disabled span .wd-icon-accent,
.emotion-1:focus:disabled span .wd-icon-accent,
.emotion-1:hover:disabled span .wd-icon-accent {
  fill: #b9c0c7;
}

.emotion-1:not([disabled]):focus,
.emotion-1:not([disabled]):active {
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

.emotion-1 span svg {
  width: 20px;
  height: 20px;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-0 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 13px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  margin-right: 24px;
}

.emotion-6 {
  display: block;
  border: none;
  padding: 0;
  margin-top: 4px;
  font-size: 13px;
  line-height: 20px;
  font-weight: 400;
  color: #0875e1;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.emotion-6:hover,
.emotion-6:active {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #0875e1;
}

.emotion-6:focus {
  background: #d7eafc;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  outline: 2px solid #d7eafc;
}

<div
  className="emotion-14"
  role="dialog"
  width={360}
>
  <div
    className="emotion-3"
  >
    <button
      aria-label="Close"
      className="emotion-1 emotion-2"
      onClick={[MockFunction]}
      title="Close"
    >
      <span
        className="emotion-0"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-x wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M5.996 4.582a.5.5 0 0 0-.717-.003l-.7.7a.504.504 0 0 0 .003.717L10.586 12l-6.004 6.004a.5.5 0 0 0-.003.717l.7.7c.197.197.514.2.717-.003L12 13.414l6.004 6.004a.5.5 0 0 0 .717.003l.7-.7a.504.504 0 0 0-.003-.717L13.414 12l6.004-6.004a.5.5 0 0 0 .003-.717l-.7-.7a.504.504 0 0 0-.717.003L12 10.586 5.996 4.582z\\" class=\\"wd-icon-fill\\"/></g></svg>",
          }
        }
      />
    </button>
  </div>
  <div
    className="emotion-13"
    width={360}
  >
    <div
      className="emotion-12"
    >
      <div
        className="emotion-10 emotion-11"
        onClose={[MockFunction]}
      >
        <span
          className="emotion-4 emotion-5"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-check wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M8.657 15.85l10.26-10.26a.49.49 0 0 1 .703.01l.7.7a.495.495 0 0 1 .01.704L9.005 18.33a.488.488 0 0 1-.692.002L8.3 18.32l-4.946-4.946a.495.495 0 0 1 .003-.711l.7-.7a.507.507 0 0 1 .711-.004l3.889 3.89z\\" class=\\"wd-icon-fill\\"/></g></svg>",
            }
          }
        />
        <div
          className="emotion-8 emotion-9"
        >
          Process failed.
          <button
            className="emotion-6 emotion-7"
            onClick={[MockFunction]}
          >
            View More
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Toast Snapshots renders with an icon 1`] = `
.emotion-12 {
  position: relative;
  width: 360px;
  -webkit-animation: animation-221pkp;
  animation: animation-221pkp;
  -webkit-animation-duration: 150ms;
  animation-duration: 150ms;
  -webkit-animation-timing-function: ease-out;
  animation-timing-function: ease-out;
  -webkit-transform-origin: top center;
  -ms-transform-origin: top center;
  transform-origin: top center;
}

.emotion-11 {
  background-color: #ffffff;
  border: 1px solid #ced3d9;
  border-radius: 3px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 16px;
  width: 360px;
}

.emotion-10 {
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-5 {
  display: inline-block;
  margin-right: 16px;
  -webkit-align-self: start;
  -ms-flex-item-align: start;
  align-self: start;
}

.emotion-5 svg {
  display: block;
}

.emotion-5 .wd-icon-fill {
  fill: #43c463;
}

.emotion-5:hover .wd-icon-fill {
  fill: #43c463;
}

.emotion-5 .wd-icon-accent {
  fill: #43c463;
}

.emotion-5:hover .wd-icon-accent {
  fill: #43c463;
}

.emotion-5 .wd-icon-background {
  fill: transparent;
}

.emotion-5:hover .wd-icon-background {
  fill: transparent;
}

.emotion-6 {
  word-break: break-word;
  word-wrap: break-word;
}

.emotion-3 {
  position: absolute;
  right: 10px;
  top: 10px;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 13px;
  border-radius: 50%;
  border: 1px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  border-width: 0px;
  background-color: transparent;
  margin: -6px;
  width: 32px;
  height: 32px;
}

.emotion-1:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-1:disabled,
.emotion-1:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-1 .wd-icon {
  display: inline-block;
  vertical-align: middle;
}

.emotion-1 span .wd-icon-fill,
.emotion-1 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #7b858f;
}

.emotion-1:focus {
  background-color: transparent;
}

.emotion-1:focus span .wd-icon-fill,
.emotion-1:focus span .wd-icon-accent {
  fill: #7b858f;
}

.emotion-1:hover:focus {
  background-color: transparent;
}

.emotion-1:active,
.emotion-1:focus:active,
.emotion-1:hover:active {
  background-color: transparent;
  border-color: transparent;
}

.emotion-1:active span .wd-icon-fill,
.emotion-1:focus:active span .wd-icon-fill,
.emotion-1:hover:active span .wd-icon-fill,
.emotion-1:active span .wd-icon-accent,
.emotion-1:focus:active span .wd-icon-accent,
.emotion-1:hover:active span .wd-icon-accent {
  fill: #333d47;
}

.emotion-1:hover {
  background-color: transparent;
}

.emotion-1:hover span .wd-icon-fill,
.emotion-1:hover span .wd-icon-accent {
  fill: #333d47;
}

.emotion-1:disabled,
.emotion-1:active:disabled,
.emotion-1:focus:disabled,
.emotion-1:hover:disabled {
  background-color: transparent;
}

.emotion-1:disabled span .wd-icon-fill,
.emotion-1:active:disabled span .wd-icon-fill,
.emotion-1:focus:disabled span .wd-icon-fill,
.emotion-1:hover:disabled span .wd-icon-fill,
.emotion-1:disabled span .wd-icon-accent,
.emotion-1:active:disabled span .wd-icon-accent,
.emotion-1:focus:disabled span .wd-icon-accent,
.emotion-1:hover:disabled span .wd-icon-accent {
  fill: #b9c0c7;
}

.emotion-1:not([disabled]):focus,
.emotion-1:not([disabled]):active {
  -webkit-animation: animation-os2b9x 100ms;
  animation: animation-os2b9x 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 2px #0875e1;
}

.emotion-1 span svg {
  width: 20px;
  height: 20px;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-0 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 13px;
  line-height: 20px;
  font-weight: 400;
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  margin-right: 24px;
}

<div
  className="emotion-12"
  role="dialog"
  width={360}
>
  <div
    className="emotion-3"
  >
    <button
      aria-label="Close"
      className="emotion-1 emotion-2"
      onClick={[MockFunction]}
      title="Close"
    >
      <span
        className="emotion-0"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-x wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M5.996 4.582a.5.5 0 0 0-.717-.003l-.7.7a.504.504 0 0 0 .003.717L10.586 12l-6.004 6.004a.5.5 0 0 0-.003.717l.7.7c.197.197.514.2.717-.003L12 13.414l6.004 6.004a.5.5 0 0 0 .717.003l.7-.7a.504.504 0 0 0-.003-.717L13.414 12l6.004-6.004a.5.5 0 0 0 .003-.717l-.7-.7a.504.504 0 0 0-.717.003L12 10.586 5.996 4.582z\\" class=\\"wd-icon-fill\\"/></g></svg>",
          }
        }
      />
    </button>
  </div>
  <div
    className="emotion-11"
    width={360}
  >
    <div
      className="emotion-10"
    >
      <div
        className="emotion-8 emotion-9"
        onClose={[MockFunction]}
      >
        <span
          className="emotion-4 emotion-5"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-check wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M8.657 15.85l10.26-10.26a.49.49 0 0 1 .703.01l.7.7a.495.495 0 0 1 .01.704L9.005 18.33a.488.488 0 0 1-.692.002L8.3 18.32l-4.946-4.946a.495.495 0 0 1 .003-.711l.7-.7a.507.507 0 0 1 .711-.004l3.889 3.89z\\" class=\\"wd-icon-fill\\"/></g></svg>",
            }
          }
        />
        <div
          className="emotion-6 emotion-7"
        >
          Success
        </div>
      </div>
    </div>
  </div>
</div>
`;
